src/backend/access/transam/varsup.c

```diff
@@ -569,6 +570,9 @@ GetNewObjectId(void)
 		}
 	}
 
+	if (GetNewObjectId_hook)
```

Comment:
this will add cost for each get oid under lock. Can we do it one time before any object is restored? may be when tsql extension is loaded?

Reply:
Eventually, we will need a hook when we make a full fix including the wraparound case.